pkg/chunk/cached_store.go

```diff
@@ -921,6 +926,20 @@ func (store *cachedStore) regMetrics(reg prometheus.Registerer) {
 		func() float64 {
 			return float64(len(store.currentUpload))
 		}))
+	reg.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name:        "store_readers",
+		Help:        "number of storage readers",
+		ConstLabels: prometheus.Labels{"storage": "cache"},
+	}, func() float64 {
+		return float64(store.cacheReaders.Load())
+	}))
+	reg.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name:        "store_readers",
+		Help:        "number of storage readers",
+		ConstLabels: prometheus.Labels{"storage": "object"},
+	}, func() float64 {
+		return float64(store.objectReaders.Load())
```

Comment:
We can add a limit for concurrent GET, and know the actual concurrency using that.

Reply:
Currently we don't have that limitation right?